core/shared/src/main/scala/zio/Runtime.scala

```diff
@@ -36,6 +37,31 @@ trait Runtime[+R] {
    */
   val platform: Platform
 
+  /**
+   * Add generic signal handler
+   *
+   * @param signal to listen for
+   * @param handler to run on receiving signal
+   */
+  def addSignalHandler(signal: String)(handler: URIO[R, Unit]): Unit =
+    Platform.addSignalHandler(signal)(() => unsafeRun(handler))
+
+  /**
+   * Use SIGUSR2 signal to interrupt runtime
+   *
+   * @param handler to run when SIGUSR2 is received
+   */
+  def addInterruptHandler(handler: URIO[R, Unit]): Unit =
```

Comment:
Maybe we should change these names to `withSignalHander`, `withInterruptHandler`, and `withFiberDumpOnInterrupt` to mirror the existing methods.